packages/slice-machine/src/features/sync/AutoSyncProvider.tsx

```diff
@@ -72,79 +73,85 @@ export const AutoSyncProvider: FC<PropsWithChildren> = (props) => {
+              isOnline,
+              authStatus,
+            });
 
-          // Now that the changes have been pushed, we need to update redux with
-          // the new remote models.
-          const serverState = await getState();
-          stableRefreshState(serverState);
-
-          // Update last sync value in local storage
-          stablePushChangesSuccess();
-        } catch (error) {
-          if (isUnauthenticatedError(error)) {
-            // If the user is not authenticated, we don't want to let the user
-            // retry the sync. We just stop the sync and let the user
-            // know that they need to login again.
-            // This can easily happen if the the token expires when the user is
-            // offline.
+          if (changedCustomTypes.length === 0 && changedSlices.length === 0) {
             return;
           }
 
-          // If the sync failed, we want to display the error message with
-          // the retry button.
-          throw error;
-        }
+          try {
+            const response = await pushChanges({
+              changedCustomTypes,
+              changedSlices,
+              // We force the deletion of documents as the auto-sync is only for
+              // the dev environment.
+              confirmDeleteDocuments: true,
+            });
+
+            // Hard limit reached is the only limit that can be reached as we set
+            // confirmDeleteDocuments to true.
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-enum-comparison
+            if (response && response.type === "HARD") {
+              console.error(
+                `Manual action required. ${
+                  response.details.customTypes.length > 1
+                    ? "These types have"
+                    : "This type has"
+                } too many associated Documents. Archive and delete these documents
+                manually and then try deleting the types again.`,
+              );
+
+              throw new Error();
+            }
+
+            // Now that the changes have been pushed, we need to update redux with
+            // the new remote models.
+            const serverState = await getState();
+            stableRefreshState(serverState);
+
+            // Update last sync value in local storage
+            stablePushChangesSuccess();
+            resolve?.();
+          } catch (error) {
+            if (isUnauthenticatedError(error)) {
+              // If the user is not authenticated, we don't want to let the user
+              // retry the sync. We just stop the sync and let the user
+              // know that they need to login again.
+              // This can easily happen if the the token expires when the user is
+              // offline.
+              return;
+            }
+
+            // If the sync failed, we want to display the error message with
+            // the retry button.
+            reject?.(error);
+            throw error;
```

Comment:
Two things here:
- above if `isUnauthenticatedError(error)`, we're not resolving/rejecting the promise, what should we do here?
- here, i don't think we want to both reject and throw, it's like bubbling the error twice

Reply:
Yeah makes sense, fixed!